From 2766b2b1f93197cca46fad1710a4ece08e2d24ff Mon Sep 17 00:00:00 2001
From: Lawrence-Tang <tangzongsheng@gmail.com>
Date: Wed, 26 Feb 2020 21:10:07 +0800
Subject: [PATCH 20/42] added support for r2s' leds, update to kernel 5.4.22

you need to update the following directory for this commit to work,
use the following commands:

  repo sync kernel
  repo sync scripts/sd-fuse
---
 .../rockchip-rk3328/base-files/etc/init.d/fa-rk3328-led    | 14 ++++++++++++++
 .../base-files/etc/modules.d/10-ledtrig-netdev             |  1 +
 .../rockchip-rk3328/base-files/etc/rc.d/S97fa-rk3328-led   |  1 +
 3 files changed, 16 insertions(+)
 create mode 100755 target/linux/rockchip-rk3328/base-files/etc/init.d/fa-rk3328-led
 create mode 100644 target/linux/rockchip-rk3328/base-files/etc/modules.d/10-ledtrig-netdev
 create mode 120000 target/linux/rockchip-rk3328/base-files/etc/rc.d/S97fa-rk3328-led

diff --git a/target/linux/rockchip-rk3328/base-files/etc/init.d/fa-rk3328-led b/target/linux/rockchip-rk3328/base-files/etc/init.d/fa-rk3328-led
new file mode 100755
index 000000000..7c720d998
--- /dev/null
+++ b/target/linux/rockchip-rk3328/base-files/etc/init.d/fa-rk3328-led
@@ -0,0 +1,14 @@
+#!/bin/sh /etc/rc.common
+
+START=97
+
+start() {
+  # eth0
+  echo netdev > /sys/class/leds/wan_led/trigger
+  echo eth0 > /sys/class/leds/wan_led/device_name
+  echo 1 > /sys/class/leds/wan_led/link
+  # eth1
+  echo netdev > /sys/class/leds/lan_led/trigger
+  echo eth1 > /sys/class/leds/lan_led/device_name
+  echo 1 > /sys/class/leds/lan_led/link
+}
diff --git a/target/linux/rockchip-rk3328/base-files/etc/modules.d/10-ledtrig-netdev b/target/linux/rockchip-rk3328/base-files/etc/modules.d/10-ledtrig-netdev
new file mode 100644
index 000000000..3b9f80505
--- /dev/null
+++ b/target/linux/rockchip-rk3328/base-files/etc/modules.d/10-ledtrig-netdev
@@ -0,0 +1 @@
+ledtrig-netdev
diff --git a/target/linux/rockchip-rk3328/base-files/etc/rc.d/S97fa-rk3328-led b/target/linux/rockchip-rk3328/base-files/etc/rc.d/S97fa-rk3328-led
new file mode 120000
index 000000000..1bdfd44d7
--- /dev/null
+++ b/target/linux/rockchip-rk3328/base-files/etc/rc.d/S97fa-rk3328-led
@@ -0,0 +1 @@
+../init.d/fa-rk3328-led
\ No newline at end of file
-- 
2.14.1.windows.1

